src/charm.py

```diff
@@ -140,73 +162,97 @@ def _update_layer(self) -> None:
             self.unit.status = BlockedStatus("Failed to replan")
```

Comment:
`BlockedStatus` should be used when we expect a human to do something, here it's not clear to me what they should do.